converter/common.py

```diff
@@ -36,3 +37,56 @@ def get_configs_json(self) -> dict:
             "info.json": str(self.info),
         }
         return configs_json
+
+
+def format_float(number: float, n: int) -> float:
+    """
+    Format float to be up to n characters wide, as precise as possible and as short
+    as possible (in descending priority). so for example given 12.333 for n=5 you will
+    get 12.33, n=7 will be 12.333
+    """
+    result = number
+    # If number is zero we just want to get 0.0 (it would mess up the log10 operation below)
+    if isclose(result, 0., rel_tol=1e-9):
+        return 0.
+
+    length = n
+
+    # Adjust length for decimal separator ('.')
+    length -= 1
+
+    # Sign messes up the log10 we use do determine how long the number is. We use
+    # abs() to fix that, but we need to remember the sign and update `n` accordingly
+    sign = 1
+
+    if result < 0:
+        result = abs(result)
```

Comment:
```suggestion
    if number < 0:
        result = abs(number)
```

better to work on original values